### ISSUE TITLE
Using only in conjunction with 2 foreign keys on a model broken

### ISSUE THREAD

**Issue by tomage** [romgar/django-dirtyfields#84]
Hello,

I found a rather interesting edge case when using this library in a project.

Namely, it seems that having a model with 2 foreign keys, and applying ".only" to it's QuerySet makes it recurse forever.

```
class A(models.Model):
    pass

class B(DirtyFieldsMixin, models.Model):
    fkey1 = models.ForeignKey(A, null=True)
    fkey2 = models.ForeignKey(A, null=True, related_name='fkey2')

def test():
    a = A.objects.create()
    b = B.objects.create(fkey1=a, fkey2=a)
    for obj in B.objects.only('fkey1'):  # RuntimeError raised! Maximum recursion depth exceeded
        print obj
```

Seems like this happens through this line in dirtyfields.py, but I didn't really dig too far into it:
```
field_value = getattr(self, field.attname)
```

**Comment by romgar** [comment]
Hi @tomage!

Thanks for the reporting.
Can you create a Pull Request which add this test (or can I use your code as an example)?
I will have a look!

**Comment by tomage** [comment]
Feel absolutely free to use my example code - should work just fine within the testing framework you already have in the library codebase (indeed that's where I tested it ;)).

Oh, one thing I completely forgot! I got this error using Django 1.9.4 - but using, say, Django 1.7 (i.e. tox -e django17), seems to work just fine.

**Comment by romgar** [comment]
You are right, it breaks from Django `1.8` series. I already had some issues with deferred fields, it seems that's not completely fixed :-)

**Comment by romgar** [comment]
This fix will be released in version `1.2.1`.
